.github/workflows/build.yml

```diff
@@ -0,0 +1,110 @@
+name: Compatibility Test
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - develop
+    - release/**
+    
+jobs:
+  tests:
+    runs-on: ${{ matrix.os }}
+    continue-on-error: false
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        
+        - version: '17.2'
+          runtime: 'iOS-17-2'
+          device: 'iPhone 15'
+          displayname: 'iPhone-15'
+          os: 'macos-14-xlarge'
+          xcode_version: '15.1'
+          needs_custom_sim: false # Takes the shipped simulator that comes with Xcode 15
+          
+        - version: '17.2'
+          runtime: 'iOS-17-2'
+          device: 'iPad Air (5th generation)'
+          displayname: 'iPad-Air-5'
+          os: 'macos-14-xlarge'
+          xcode_version: '15.1'
+          needs_custom_sim: false # Takes the shipped simulator that comes with Xcode 15
+        
+        - version: '16.4'
+          runtime: 'iOS-16-4'
+          device: 'iPhone 14'
+          displayname: 'iPhone-14'
+          os: 'macos-13'
+          xcode_version: '14.3'
+          needs_custom_sim: false # Takes the shipped simulator that comes with Xcode 14
+        
+      # Github deprecated macos-12 runners
```

Comment:
Still needed?

Reply:
I think it's still good to remind ourselves that we want to add them again in the future